src/main/java/pl/antonina/tasks/child/ChildService.java

```diff
@@ -37,8 +37,8 @@ void addChild(long parentId, ChildData childData) {
         Parent parent = parentRepository.findById(parentId).orElseThrow();
 
         User user = new User();
-        user.setEmail(childData.getEmail());
-        user.setPassword(childData.getPassword());
+        user.setEmail(childData.getUserData().getEmail());
+        user.setPassword(childData.getUserData().getPassword());
```

Comment:
This is the common part for User, Child and Parent. It would look much more cleaner to add a method that can set these two fields getting UserData as the argument. It would also hide the implementation details (it seems that this service only needs to know it must set user data, not what is included in these details). It will allow you to add more variables to user data later with much less hassle.
http://wiki3.cosc.canterbury.ac.nz/index.php/Message_chain_smell
https://refactoring.guru/smells/message-chains

Reply:
Done :)